kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller.go

```diff
@@ -678,7 +774,66 @@ func (r *TaskRunToolCallReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, err
 		}
 
+		approvalGranted := false
+		trtcNamespace := trtc.Namespace
+		var contactChannel *kubechainv1alpha1.ContactChannel
+		var apiKey string
+
 		if needsApproval {
+			contactChannel, err = r.getContactChannel(ctx, mcpServer, trtcNamespace)
+			if err != nil {
+				return r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallStatusTypeErrorRequestingHumanApproval, "NoContactChannel", &trtc, err)
+			}
+
+			apiKey, err = r.getHumanLayerAPIKey(ctx, contactChannel.Spec.APIKeyFrom.SecretKeyRef.Name, contactChannel.Spec.APIKeyFrom.SecretKeyRef.Key, trtcNamespace)
+
+			if err != nil || apiKey == "" {
+				return r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallStatusTypeErrorRequestingHumanApproval, "NoAPIKey", &trtc, err)
+			}
+		}
+
+		// Check if approved/rejected
+		if needsApproval && trtc.Status.Status == kubechainv1alpha1.TaskRunToolCallStatusTypeAwaitingHumanApproval {
+			client := r.HLClient.NewHumanLayerClient()
+			client.SetCallID(trtc.Status.HumanLayerCallId)
+			client.SetAPIKey(apiKey)
```

Comment:
I think the way they want you to do this might be 

```go
functionCall, _, _ := r.HLClient.NewHumanLayerClient().
  SetCallID(...).
  SetAPIKey(...).
  GetFunctionCallStatus(ctx)

```

they might be wrong though. nothing wrong with this pattern here